File: nebras-app/app/src/main/kotlin/com/nebras/mobile/feature/home/view/AboutScreen.kt
package com.nebras.mobile.feature.home.view

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.PrivacyTip
import androidx.compose.material.icons.rounded.MenuBook
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Scaffold
import androidx.compose.material3.SnackbarHost
import androidx.compose.material3.SnackbarHostState
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.navigation.NavHostController
import coil3.compose.AsyncImage
import coil3.request.ImageRequest
import com.nebras.mobile.BuildConfig
import com.nebras.mobile.core.theme.NebrasTextStyles
import com.nebras.mobile.core.widget.AppBarWidget
import com.nebras.mobile.core.widget.openExternalUrlSafely
import kotlinx.coroutines.launch

/** رابط سياسة الخصوصية العامّ (بدون تسجيل دخول — متوافق مع Google Play). */
const val PRIVACY_POLICY_URL = "https://nibras-app-website.vercel.app/privacy"

/**
 * شاشة «حول» — نقل `features/home/view/about_screen.dart`.
 * تعرض اسم التطبيق ورسالته وسياسة المحتوى ورابط الخصوصية ورقم الإصدار.
 */
@Composable
fun AboutScreen(navController: NavHostController) {
    val context = LocalContext.current
    val scope = rememberCoroutineScope()
    val snackbarHostState = remember { SnackbarHostState() }

    // يُقرأ من حزمة التطبيق وقت التشغيل بدل رقم ثابت قد يتعارض مع الإصدار
    // الفعليّ (بديل `PackageInfo.fromPlatform()`).
    val version = remember { BuildConfig.VERSION_NAME }

    Scaffold(
        containerColor = MaterialTheme.colorScheme.background,
        snackbarHost = { SnackbarHost(snackbarHostState) },
        topBar = {
            AppBarWidget(
                title = "حول",
                onBack = { navController.popBackStack() },
            )
        },
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .verticalScroll(rememberScrollState())
                .padding(24.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            Spacer(Modifier.height(40.dp))
            AppIcon()
            Spacer(Modifier.height(24.dp))
            // اسم التطبيق
            Text("نبراس", style = NebrasTextStyles.heading1)
            Spacer(Modifier.height(8.dp))
            // العبارة التعريفيّة
            Text(
                text = "بيئة معرفية متكاملة للقارئ العربي",
                modifier = Modifier.fillMaxWidth(),
                style = NebrasTextStyles.body,
                textAlign = TextAlign.Center,
            )
            Spacer(Modifier.height(40.dp))
            // رسالتنا
            Text(
                text = "رسالتنا",
                modifier = Modifier.fillMaxWidth(),
                style = NebrasTextStyles.heading3,
                textAlign = TextAlign.Start,
            )
            Spacer(Modifier.height(12.dp))
            Text(
                text = "نبراس منصّة معرفية عربية حديثة، غايتنا أن نقدّم للقارئ العربي " +
                    "محتوى جميلاً وراقياً في بيئة واحدة متكاملة: كتب ومواد صوتية " +
                    "ومرئية ومقالات تنمو وتتجدّد يوماً بعد يوم، لتيسير الوصول إلى " +
                    "المعرفة بلغة عربية خالصة. محتوانا معرفيّ عام لا يقتصر على دين " +
                    "أو طائفة أو توجّه — فالمعرفة هنا للجميع.\n\n" +
                    "والنشر في نبراس ليس حكراً علينا: عبر ميزة «القنوات» يمكن لأيّ " +
                    "مستخدم أن ينشئ قناته الخاصة ويشارك الجمهور ما لديه من محتوى " +
                    "ومعرفة من داخل التطبيق مباشرة.",
                modifier = Modifier.fillMaxWidth(),
                style = NebrasTextStyles.body,
                textAlign = TextAlign.Start,
            )
            Spacer(Modifier.height(32.dp))
            // سياسة المحتوى — تصريح متوافق مع Google Play بأنّ التطبيق لا ينشر
            // عمداً محتوًى مخالفاً، مع الإشارة لإمكانيّة الإبلاغ.
            Text(
                text = "سياسة المحتوى",
                modifier = Modifier.fillMaxWidth(),
                style = NebrasTextStyles.heading3,
                textAlign = TextAlign.Start,
            )
            Spacer(Modifier.height(12.dp))
            Text(
                text = "نبراس منصّة محتوى موسوعيّ معرفيّ عامّ. لا ننشر عمداً أيّ محتوى " +
                    "ينتهك حقوق الملكية الفكرية أو حقوق النشر، ولا أيّ محتوى جنسيّ صريح " +
                    "أو يضرّ بالأطفال أو يحضّ على العنف أو الإرهاب. كلّ المحتوى يُفترض أن " +
                    "يكون عامّاً أو مرخّصاً. إن صادفت مخالفة، استخدم زرّ «إبلاغ» على " +
                    "المحتوى وسنراجعها ونزيلها فوراً عند ثبوتها.",
                modifier = Modifier.fillMaxWidth(),
                style = NebrasTextStyles.body,
                textAlign = TextAlign.Start,
            )
            Spacer(Modifier.height(32.dp))
            // سياسة الخصوصية — رابط عامّ يُفتح في المتصفّح الخارجيّ.
            OutlinedButton(
                onClick = {
                    val ok = openExternalUrlSafely(context, PRIVACY_POLICY_URL)
                    if (!ok) {
                        scope.launch { snackbarHostState.showSnackbar("سياسة الخصوصية") }
                    }
                },
            ) {
                Icon(Icons.Outlined.PrivacyTip, contentDescription = null)
                Spacer(Modifier.width(8.dp))
                Text("سياسة الخصوصية")
            }
            Spacer(Modifier.height(24.dp))
            // الإصدار — يُعرض فقط بعد قراءته من الحزمة (لا رقم ثابت).
            if (version.isNotEmpty()) {
                Text(
                    text = "الإصدار $version",
                    style = NebrasTextStyles.greysmallText,
                )
            }
            Spacer(Modifier.height(24.dp))
        }
    }
}

/**
 * أيقونة التطبيق من الأصول، ومع تعذّر تحميلها مربّع بلون أساسيّ شفّاف يحمل
 * أيقونة كتاب (نظير `errorBuilder` في Flutter).
 */
@Composable
private fun AppIcon() {
    val context = LocalContext.current
    var failed by remember { mutableStateOf(false) }
    val shape = RoundedCornerShape(20.dp)

    if (failed) {
        Box(
            modifier = Modifier
                .size(96.dp)
                .clip(shape)
                .background(MaterialTheme.colorScheme.primary.copy(alpha = 0.1f)),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                Icons.Rounded.MenuBook,
                contentDescription = null,
                modifier = Modifier.size(40.dp),
                tint = MaterialTheme.colorScheme.primary,
            )
        }
        return
    }

    AsyncImage(
        // نفس أصل Flutter: assets/images/app_icon.png (نُسِخ إلى assets/).
        model = ImageRequest.Builder(context)
            .data("file:///android_asset/images/app_icon.png")
            .build(),
        contentDescription = null,
        contentScale = ContentScale.Crop,
        onError = { failed = true },
        modifier = Modifier
            .size(96.dp)
            .clip(shape),
    )
}

File: app-flutter/lib/features/home/view/about_screen.dart
import 'package:easy_localization/easy_localization.dart';
import 'package:flutter/material.dart';
import 'package:flutter_screenutil/flutter_screenutil.dart';
import 'package:nebras_mobile_app/core/extensions/sized_boxextension.dart';
import 'package:nebras_mobile_app/core/theme/text_styles.dart';
import 'package:nebras_mobile_app/core/widgets/app_bar_widget.dart';
import 'package:package_info_plus/package_info_plus.dart';
import 'package:url_launcher/url_launcher.dart';

/// رابط سياسة الخصوصية العامّ (بدون تسجيل دخول — متوافق مع Google Play).
const String kPrivacyPolicyUrl = 'https://nibras-app-website.vercel.app/privacy';

/// About Screen
/// Shows app name, mission, and version info
class AboutScreen extends StatefulWidget {
  const AboutScreen({super.key});

  @override
  State<AboutScreen> createState() => _AboutScreenState();
}

class _AboutScreenState extends State<AboutScreen> {
  // يُقرأ من حزمة التطبيق وقت التشغيل بدل رقم ثابت قد يتعارض مع الإصدار الفعليّ.
  String _version = '';

  @override
  void initState() {
    super.initState();
    _loadVersion();
  }

  Future<void> _loadVersion() async {
    try {
      final info = await PackageInfo.fromPlatform();
      if (mounted) setState(() => _version = info.version);
    } catch (_) {
      /* في حال تعذّر القراءة نُبقي الحقل فارغاً فلا يظهر رقم خاطئ. */
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Theme.of(context).scaffoldBackgroundColor,
      appBar: AppBarWidget(title: 'About'.tr()),
      body: SafeArea(
        child: SingleChildScrollView(
          padding: EdgeInsets.all(24.w),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              40.sbh,
              ClipRRect(
                borderRadius: BorderRadius.circular(20.r),
                child: Image.asset(
                  'assets/images/app_icon.png',
                  width: 96.w,
                  height: 96.w,
                  fit: BoxFit.cover,
                  errorBuilder: (_, _, _) => Container(
                    width: 96.w,
                    height: 96.w,
                    decoration: BoxDecoration(
                      color: Theme.of(
                        context,
                      ).colorScheme.primary.withValues(alpha: 0.1),
                      borderRadius: BorderRadius.circular(20.r),
                    ),
                    child: Icon(
                      Icons.menu_book_rounded,
                      size: 40.sp,
                      color: Theme.of(context).colorScheme.primary,
                    ),
                  ),
                ),
              ),
              24.sbh,
              // App name
              Text('Nebras'.tr(), style: TextStyles.heading1(context)),
              8.sbh,
              // Tagline
              Text(
                'Explore the archives of knowledge'.tr(),
                style: TextStyles.body(context),
                textAlign: TextAlign.center,
              ),
              40.sbh,
              // Mission section
              Align(
                alignment: AlignmentDirectional.centerStart,
                child: Text(
                  'Our Mission'.tr(),
                  style: TextStyles.heading3(context),
                ),
              ),
              12.sbh,
              Text(
                'About Mission Text'.tr(),
                style: TextStyles.body(context),
                textAlign: TextAlign.start,
              ),
              32.sbh,
              // سياسة المحتوى — تصريح متوافق مع Google Play بأنّ التطبيق
              // لا ينشر عمداً محتوًى مخالفاً، مع الإشارة لإمكانيّة الإبلاغ.
              Align(
                alignment: AlignmentDirectional.centerStart,
                child: Text(
                  'content_policy_title'.tr(),
                  style: TextStyles.heading3(context),
                ),
              ),
              12.sbh,
              Text(
                'content_policy_text'.tr(),
                style: TextStyles.body(context),
                textAlign: TextAlign.start,
              ),
              32.sbh,
              // Privacy Policy — رابط عامّ يُفتح في المتصفّح الخارجيّ.
              OutlinedButton.icon(
                onPressed: () => _openPrivacyPolicy(context),
                icon: const Icon(Icons.privacy_tip_outlined),
                label: Text('Privacy Policy'.tr()),
              ),
              24.sbh,
              // Version — يُعرض فقط بعد قراءته من الحزمة (لا رقم ثابت).
              if (_version.isNotEmpty)
                Text(
                  '${'Version'.tr()} $_version',
                  style: TextStyles.greysmallText(context),
                ),
              24.sbh,
            ],
          ),
        ),
      ),
    );
  }

  Future<void> _openPrivacyPolicy(BuildContext context) async {
    final uri = Uri.parse(kPrivacyPolicyUrl);
    final ok = await launchUrl(uri, mode: LaunchMode.externalApplication);
    if (!ok && context.mounted) {
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(content: Text('Privacy Policy'.tr())),
      );
    }
  }
}
